.github/workflows/auto-formatter.yml

```diff
@@ -0,0 +1,21 @@
+name: gofumpt
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+        fetch-depth: 0
+
+    - name: Auto Format code
+      uses: iamnotaturtle/auto-gofmt@v2.0.0
```

Comment:
Thanks for the PR, but I'd prefer not introducing a 3rd party github action for formatting.

Reply:
Hey @mfridman does that mean the regular `goimports` or `go fmt` would be preferred here?

Reply:
Ye, most projects I've seen have linting, formatting, etc. run against the pull request itself. If there is a diff or something doesn't match it is up to the author to fix (CI fails).

I don't have a strong opinion on whether to use `gofmt` or `gofumpt`. The latter has [pre-built binaries](https://github.com/mvdan/gofumpt/releases/tag/v0.3.1) so can curl it.

Something like this can be added before the tests:

```
- name: Check Go code formatting
  run: |
    if [ "$(gofmt -s -l . | wc -l)" -gt 0 ]; then
      gofmt -s -l .
      echo "Please format Go code by running: go fmt ./..."
    exit 1
    fi
```

This avoids a dependency on a GitHub Action and enforces standard PR's to make reviews easier.

Reply:
I think the CI failing actually is the way go. I have it failing on all of my other projects. Usually author is responsible to fix them. I'll go ahead and make the PR changes :)